src/Core/Validation/FieldValidation/YearFieldValidator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace SilverStripe\Core\Validation\FieldValidation;
+
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Core\Validation\FieldValidation\NumericFieldValidator;
+
+/**
+ * Validates that a field is an integer year between two dates, or 0 for a null value.
+ */
+class YearFieldValidator extends IntFieldValidator
```

Comment:
It's all because MySQL YEAR is a really weird thing has this restrictive range range from 1901 to 2155, though 0 is also valid.  Means we can't just use IntFieldValidator with a minValue of 1901